File: Data Structures & Algorithms/minimum-stack/submission-0.cs
public class MinStack {
    int[] stack;
    int length = 0;
    int minVal;

    public MinStack() {
        stack = new int[2];
    }
    
    public void Push(int val) {
        if ((length + 1) == stack.Length)
        {
            int[] temp = new int[stack.Length * 2];

            for (int i = 0; i < stack.Length; i++)
            {
                temp[i] = stack[i];
            }

            stack = temp;
        }

        stack[length] = val;
        length++;
    }
        
    
    public void Pop() {
        length--;
    }
    
    public int Top() {
        return stack[length - 1];
    }
    
    public int GetMin() {
        minVal = stack[0];

        for (int i = 0; i < length; i++)
        {
            if (stack[i] < minVal)
                minVal = stack[i];
        }

        return minVal;
    }
}

File: Data Structures & Algorithms/baseball-game/submission-0.cpp
class Solution {
public:
    int calPoints(vector<string>& operations) {
        stack<int> _stack;

        for(int i=0; i< operations.size(); i++) {
            string var = operations[i];

            if(var == "+") {
                int nums[2];
                stack<int> tempStack(_stack);

                for (int i=0; i < 2; i++) {
                    nums[i] = tempStack.top();
                    tempStack.pop();
                }

                _stack.push(nums[0] + nums[1]);
            }
            else if(var == "D") {
                _stack.push(_stack.top() * 2);
            }
            else if(var == "C") {
                _stack.pop();
            }
            else {
                _stack.push(stoi(var));
            }
        }

        int returnNum = 0;

        while(!_stack.empty()) {
            returnNum += _stack.top();
            _stack.pop();
        }

        return returnNum;
    }
};